### ISSUE TITLE
Strange Node View behavior for optional segments on CLP that have been enabled and then disabled

### ISSUE THREAD

**Issue by davidmpickett** [department-of-veterans-affairs/va.gov-cms#19165]
## Statuses
[2024-09-12] [Fran] Decision log added identifying Option 1 as the desired solution; AC added; moved to the bottom of Next Refinement.

## Description
When sections are unchecked/disabled, clear out any text/values previously entered.

## Background
This came up during Q&A for #18954 and was deemed out of scope.

Note that there are two distinct, but related related errors you might encounter while reproducing this issue
- #19161
- #19163

## Reproduction steps
1. Go to staging.cms.va.gov (or a Tugboat)
2. [Create a new Campaign Landing Page](https://staging.cms.va.gov/node/add/campaign_landing_page)
   - [ ] Only fill in the required fields in the Hero banner, Why this matters, What you can do, and VA benefits sections
   - [ ] Save the node
   - [ ] Validate that the Node View matches the fields populated on Node Edit
 3. Edit the node and add an optional section
    - [ ] Enable one of the following optional sections, Video, Spotlight, Stories, Downloadable resources, Events, FAQs
    - [ ] Fill out required fields for the optional section 
    - [ ] Save the node
    - [ ] Validate that the Node View updates with the new section
4. Edit the node and disable the optional section
    - [ ] Uncheck the "Enable this page segment" box on the section you just added
    - [ ] Save the node
    - [ ] Notice that Node View still shows the content from the subfields of the section you disabled  

## Product-preferred solution path (previously: Option 1)
- We should clear out any content entered the sub-fields (on Save?) if the enabled box is unchecked
- Consider the **Banner alert** fieldset on VBA Facility content type. It clears out all content in sub-fields once the "Display a banner alert on this facility box" is unchecked and the node is saved.

## Decision Log
Option 1 will be implemented, mimicking existing CMS functionality (e.g. Events).

### Scope / Impact analysis
_What, if anything, could break as a result of this change?_
_Engineer should assess this when approaching PR._

### Roles / assignments
_After functional testing, code review, accessibility review, and design review can happen in parallel._
* **Functional testing:** 
* **Code review:** 
* **Accessibility review:** @laflannery 
* **Product acceptance:** @FranECross 

## Acceptance Criteria
- [ ] Editing a Campaign Landing Page that has one or more optional sections enabled (e.g. Video, Spotlight, Stories, Downloadable resources, Events, FAQsA) by unchecking an optional section results in clearing out the previously entered content (mimicking current behavior in CMS e.g. events)
- [ ] Does this product have an existing regression test plan? No, regression test plan will be created as part of the work / points in this ticket
- [ ] Accessibility review

**Comment by FranECross** [comment]
@davidmpickett @dsasser I'm looking at the options and thinking #2 would be a good solution, IF we have situations where an editor might follow the following scenario when entering information: They enter information into a CLP that they want to publish right away, but info they enter into a sub-field they want to get further approval on. They uncheck the box so it doesn't display, but they don't want their information to be cleared out because when they get approval, they will then edit, uncheck the box so that it WILL display, and then republish. Thoughts on feasibility and effort?

**Comment by dsasser** [comment]
In other areas of the CMS, Events for instance, we clear the data from fields that are hidden when making form selections.

> They enter information into a CLP that they want to publish right away, but info they enter into a sub-field they want to get further approval on. They uncheck the box so it doesn't display, but they don't want their information to be cleared out because when they get approval, they will then edit, uncheck the box so that it WILL display, and then republish. Thoughts on feasibility and effort?

I would use a node revision in this case. Here the user would create the publish-able version and publish it. Then create a draft of the new changes that need approval.

In my opinion we should (always?) be clearing data when it is hidden dynamically.

**Comment by davidmpickett** [comment]
I agree with @dsasser that option 1 / clearing out the data is generally the more robust/performant option with dynamically hidden fields. It prevents weird validation glitches like #19161. It's also an established pattern within the CMS so would be relatively straightforward to implement. 

Option 2 would be a whole new paradigm and would in someways us fighting against the grain to accommodate a small, hypothetical use case

There's also Option 0, which is to leave it as is. This is not breaking any functionality, just leading to possible confusion.

**Comment by FranECross** [comment]
- [ ] When creating a Campaign Landing Page with one or more optional sections enabled, and during creation an optional section is unchecked, the content is cleared out. (mimicking current behavior in CMS e.g. events)

@FranECross to create a spike to look at other fields and this ticket is restricted to this particular issue.

**Comment by dsasser** [comment]
## Findings

BLUF: This issue is a no-op.

As discussed in scrum, the issue with data from unselected panels/segments (video, spotlight, etc) appearing on Drupal node view pages is complex to resolve, mostly due to the nested paragraph situations we have in some segments. In order to resolve this on the Drupal side would take significant engineering effort. As a result, we determined that we should confirm if there were any controls on the FE that would prevent showing panel information for unselected panels. If such controls were in place, and there is no risk to exposing unwanted data to the FE, this issue would become a no-op.

On the FE, in content-build, the CLP templates all have a conditional check to determine if the panel/segment is enabled first before rendering any of the panel's fields. This results in preventing the unwanted data from appearing in unselected panels. Here are a few screenshots demonstrating the safety mechanisms:

![Image](https://github.com/user-attachments/assets/89cfe9d5-cd11-451c-9aa4-a5c3348e6963)
![Image](https://github.com/user-attachments/assets/963e72d4-525d-4b8c-a039-d8cdf764b1f2)
![Image](https://github.com/user-attachments/assets/699e5c36-e295-470b-a540-57588b117cd3)
![Image](https://github.com/user-attachments/assets/e19a27a4-b368-447d-8a92-f39128439a8f)
![Image](https://github.com/user-attachments/assets/a3a1e55b-ecb4-4eea-b0da-bb42e42d7f88)
![Image](https://github.com/user-attachments/assets/c437f843-438a-4f8f-9bb8-752fca6f51c9)
